projects/observability/src/shared/components/cartesian/d3/legend/cartesian-legend.ts

```diff
@@ -133,4 +183,33 @@ export class CartesianLegend {
       })
     );
   }
+
+  private resetToDefault(): void {
+    this.activeSeries = [...this.initialSeries];
+    this.isDefault = true;
+    this.updateLegendTextClasses();
+    this.toggleReset(this.isDefault);
+    this.activeSeriesSubject.next();
+  }
+
+  private updateActiveSeries(seriesEntry: Series<{}>): void {
+    if (this.isDefault) {
+      this.activeSeries = [];
+      this.activeSeries.push(seriesEntry);
+      this.isDefault = false;
+    } else {
+      if (this.isThisLegendEntryActive(seriesEntry)) {
+        this.activeSeries = this.activeSeries.filter(series => series !== seriesEntry);
+      } else {
+        this.activeSeries.push(seriesEntry);
+      }
+    }
+    this.updateLegendTextClasses();
+    this.toggleReset(this.isDefault);
+    this.activeSeriesSubject.next();
+  }
+
+  private isThisLegendEntryActive(seriesEntry: Series<{}>): boolean {
+    return this.activeSeries.findIndex(series => series === seriesEntry) >= 0;
```

Comment:
nit: `contains` (or something like that - there's a simple util in lodash)

Reply:
used `includes`